app/src/main/java/com/example/poste/activities/FolderViewActivity.java

```diff
@@ -72,77 +50,85 @@ protected void onCreate(Bundle savedInstanceState) {
+        });
+
+        // Create listeners for the post buttons
+        openBut.setOnClickListener(view -> {
+            if (PosteApplication.getSelectedPost() != null) {
+                try {
+                    Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(PosteApplication.getSelectedPost().getLink()));
+                    startActivity(browserIntent);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    Toast.makeText(FolderViewActivity.this, R.string.internal_error, Toast.LENGTH_LONG).show();
+                }
             }
+        });
 
-            // Fill post view (Recycler View)
-            postRecyclerView.setLayoutManager(new LinearLayoutManager(FolderViewActivity.this));
-            postAdapter = new PostAdapter(
-                    new PostAdapter.ClickListener() {
-                        @Override
-                        public void onItemClick(int position, Post post) {
-                            try {
-                                Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(post.getLink()));
-                                startActivity(browserIntent);
-                            } catch (Exception e) {
-                                e.printStackTrace();
-                                Toast.makeText(FolderViewActivity.this, R.string.internal_error, Toast.LENGTH_LONG).show();
-                            }
-                        }
-
-                        @Override
-                        public void onItemLongClick(int position, Post model) {
-//                            Toast.makeText(FolderViewActivity.this, "Share this folder", Toast.LENGTH_LONG).show();
-                        }
-                    },
-                    currentFolder.getPosts()
-            );
-            postRecyclerView.setAdapter(postAdapter);
-            registerForContextMenu(postRecyclerView);
-        } catch (APIException e) {
-            throw new RuntimeException(e);
-        }
-    }
+        editBut.setOnClickListener(view -> {
+            if (PosteApplication.getSelectedPost() != null) {
+                Intent intent = new Intent(FolderViewActivity.this, EditPostActivity.class);
+                startActivity(intent);
+            }
+        });
 
-    @Override
-    public boolean onContextItemSelected(MenuItem item) {
-        Post post = postAdapter.getLocalDataSetItem();
-        Intent intent = null;
-        switch (item.getItemId()) {
-            case R.id.ctx_menu_edit_post:
-                intent = new Intent(FolderViewActivity.this, EditPostActivity.class);
-                intent.putExtra("folderId", currentFolder.getId());
-                intent.putExtra("postId", post.getId());
+        deleteBut.setOnClickListener(view -> {
+            if (PosteApplication.getSelectedPost() != null) {
+                PosteApplication.getSelectedFolder().removePost(PosteApplication.getSelectedPost());
```

Comment:
I like the method call here to abstract functionality, but I think removePost should also handle an API call (see the comment above on adding a post) for the same reasons.

Reply:
Discussed below.